internal/linux/linux.go

```diff
@@ -0,0 +1,72 @@
+package linux
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// Dup3 wraps [unix.Dup3].
+func Dup3(oldfd, newfd, flags int) error {
+	return os.NewSyscallError("dup3", retryOnEINTR(func() error {
+		return unix.Dup3(oldfd, newfd, flags)
```

Comment:
nit: I think splitting this in more lines can be clearer. The functio that takes several params, the third of which is another function that takes a function param that is defined inline is a little bit too much, IMHO.

Reply:
Fixed this and also Sendmsg in the same way.